src/Symfony/Component/Yaml/Parser.php

```diff
@@ -111,6 +137,8 @@ public function parse($value, $flags = 0)
             $data = $this->doParse($value, $flags);
         } catch (\Exception $e) {
         } catch (\Throwable $e) {
+        } finally {
+            $this->filename = null;
```

Comment:
It's strange to set the property in `parseFile` and reset it here. Why not adding a try finally in the `parseFile` method instead ?